src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -265,6 +309,16 @@ public void onExit() {
                 .filter(e -> !e.getValue().isCommitted())
                 .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().getSentOffset()));
 
+        final String appName = getContext().getParameters().getConsumingAppId();
+        final NakadiKpiPublisher kpiPublisher = getContext().getKpiPublisher();
+        final Subscription subscription = getContext().getSubscription();
+        subscription.getEventTypes().stream()
+                .forEach(et -> {
+                    final long bytes = kpiDataPerEventType.get(et).getAndResetBytesSent();
+                    final long count = kpiDataPerEventType.get(et).getAndResetNumberOfEventsSent();
+                    publishKpi(appName, kpiPublisher, et, bytes, count);
+                });
```

Comment:
This code is very similar to the code you have some lines above. Maybe it is possible to make a function from it and use it in 2 places? Also, here you don't check if count is zero.

Reply:
I wanted to count at least once per stream. Even if no events were transferred. This way we would not miss zombie consumers from the count, since they are actually using the system.

Ok, I'll try to improve the reutilisation.

Reply:
Ok I see now. For me it was just strange that you count empty batches data in one case and don't count in another.

Reply:
Improved a little https://github.com/zalando/nakadi/pull/813/commits/3ceea68d643556a8572e43c1c7ee1018d36e960c